lib/impress.files.js

```diff
@@ -69,19 +66,18 @@ Client.prototype.upload = function(
   done // optional callback function(err, doneCount)
 ) {
   const client = this;
+  done = api.metasync.cb(done);
 
   let fileCount = 0;
   let doneCount = 0;
 
   const cb = (err, data) => {
     doneCount++;
     if (each) each(err, data);
-    if (fileCount === doneCount && done) done(null, doneCount);
+    if (fileCount === doneCount) done(null, doneCount);
   };
 
-  if (!client.files) {
-    if (done) return done(null, 0);
-  }
+  if (!client.files) return done(null, 0);
```

Comment:
Can be moved to the function beginning as "guard precondition" + avoid redundant variables declarations